config/main.py

```diff
@@ -524,6 +524,25 @@ def load_minigraph():
     click.echo("Please note setting loaded from minigraph will be lost after system reboot. To preserve setting, run `config save`.")
 
 
+#
+# 'hostname' command
+#
+@config.command('hostname')
+@click.argument('new_hostname', metavar='<new_hostname>', required=True)
+def hostname(new_hostname):
+    """Change device hostname without impacting the traffic."""
+
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    config_db.mod_entry('DEVICE_METADATA' , 'localhost', {"hostname" : new_hostname})
```

Comment:
I've one question... What if 'localhost' is not present in 'DEVICE_METADATA'? Does this create a new one?

Reply:
I verified this thing, @prsunny. It will add the "hostname" key to the "DEVICE_METADATA|localhost" field in config DB.